Seagate/cortx-k8s
# CORTX-34139: Remove all deprecated code

## Description

In v0.12.0 we announced future removal of deprecated code and Kubernetes resources for v0.13.0. This change is the removal of said code/resources.

- Remove deletion of deprecated Charts and resources in uninstall script
- Remove commented out Consul patching from install script
- Remove `cortx-control-loadbal-svc` Service
- Remove deprecated `cortx-io-svc-N` Service
- Remove filters for deprecated services in service status script

## Breaking change

- Removal of Services `cortx-control-loadbal-svc` and `cortx-io-svc-N`. Use `cortx-control` and `cortx-server-N` Service names instead.

In general, running the utility scripts from a new release against clusters deployed from previous versions of cortx-k8s is not recommended, but may work. In this case, with 0.13.0:

- Running the uninstall script for a cluster deployed with v0.9.0 or earlier will fail to cleanup all resources. The solution is to either uninstall the cluster prior to using the new script version, or manually cleanup the left-over resources. It should work with v0.10.0 to v0.12.0 deployed clusters, but has not been tested.
- Running the status script on an v0.12.0 or earlier deployed cluster may erroneously report issues due to the existence of the deprecated services.

## Type of change

- [ ] New feature (non-breaking change that adds new functionality)
- [ ] Breaking change (bug fix or new feature that breaks existing functionality)
- [ ] Third-party dependency update
- [ ] Documentation additions or improvements
- [X] Code quality improvements to existing code or test additions/updates

## Applicable issues

- This change fixes an issue: CORTX-34139

## How was this tested?

- Deploy cluster
- Run S3 I/O after deployment
- Run CSM I/O after deployment
- Destroy cluster, notice no leftover resources
- Observed expected services, and only those, are installed
- Observed no items exist with label `cortx.io/deprecated=true`

## Additional information

- As a result of removing deprecated code, other dead code was removed.
- Did some misc cleanup of bash code
- Fixed a bug with secret generation if the solution.yaml file field had an in-line comment
- Secret temporary files are in a direct subdirectory, not in the obsolete `cortx-cloud-helm-pkg` directory
- Remove some dependence on custom YAML parsing functions and scripts, use yq directly
- Parallelize deletion of PVCs using Chart-based labels. TODO: see if we can tag Consul PVCs with the same label to further parallelize all PVC deletions.
- Make output of destroy script a bit less verbose
- Changed the order of resource uninstallation to match the reverse of installation. This shouldn't have much noticeable effect.

## Checklist

- [X] The change is tested and works locally.
- [ ] New or changed settings in the solution YAML are documented clearly in the README.md file.
- [X] All commits are signed off and are in agreement with the [CORTX Community DCO and CLA policy](https://github.com/Seagate/cortx/blob/main/doc/dco_cla.md).

If this change requires newer CORTX or third party image versions:

- [ ] The `image` fields in [solution.example.yaml](../k8_cortx_cloud/solution.example.yaml) have been updated to use the required versions.
- [ ] The `appVersion` field of the [Helm chart](../charts/cortx/Chart.yaml) has been updated to use the new CORTX version.

If this change addresses a CORTX Jira issue:

- [X] The title of the PR starts with the issue ID (e.g. `CORTX-XXXXX:`)
